pkg/controller/bootimage/capi_helpers.go

```
@@ -0,0 +1,647 @@
+		return fmt.Errorf("unable to patch CAPI MachineDeployment %s: %w", md.Name, err)
+	}
+	klog.Infof("Successfully patched CAPI MachineDeployment %s to use infrastructure template %s", md.Name, newTemplateName)
+	return nil
+}
+
+// newInfraTemplateName returns a deterministic name for a new infrastructure template.
+// Format: <machineset-name>-<hash8> where hash8 is the 8 hex digit FNV-1a 32-bit hash of
+// the new template spec. Matches the naming convention used by cluster-capi-operator.
+func newInfraTemplateName(msName string, newSpec []byte) string {
+	hasher := fnv.New32a()
+	hasher.Write(newSpec)
```

golangci-lint (2.12.2)

[low] G104: Errors unhandled

(gosec)